File: examples/lua/src/greeting.lua
local M = {}

function M.message(name)
  if name == "" then
    return "Hello!"
  end

  return "Hello, " .. name .. "!"
end

function M.audience(count)
  if count == 1 then
    return "person"
  end

  return "people"
end

return M

File: examples/cpp/src/greeting.cpp
#include "greeting.hpp"

std::string greeting(const std::string &name) {
  if (name.empty()) {
    return "Hello!";
  }

  return "Hello, " + name + "!";
}

std::string audience(int count) {
  if (count == 1) {
    return "person";
  }

  return "people";
}
